### PR TITLE
tests/resource/guardduty_filter: Fix hardcoded region

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #12995 [Phase 2](https://github.com/hashicorp/terraform-provider-aws/issues/12995#issuecomment-730680494)

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

### Acceptance Tests in GovCloud

```
--- PASS: TestAccAWSGuardDuty_serial (116.42s)
    --- PASS: TestAccAWSGuardDuty_serial/Filter (116.42s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/basic (31.76s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/update (29.54s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/tags (40.17s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/disappears (14.95s)
```

### CODE REVIEW

File: aws/resource_aws_guardduty_filter_test.go [hunk 1]
```diff
@@ -257,7 +257,7 @@ resource "aws_guardduty_filter" "test" {
   finding_criteria {
     criterion {
       field  = "region"
-      equals = ["eu-west-1"]
```
Comment (bflad) [inline aws/resource_aws_guardduty_filter_test.go:260]: Similar note here and below about potentially using `aws_region` data source instead (the check can be updated to `testAccGetRegion()` for simplicity.

**Comment by YakDriver** [comment]
It's happy as a clam:

GovCloud:
```
--- PASS: TestAccAWSGuardDuty_serial (116.12s)
    --- PASS: TestAccAWSGuardDuty_serial/Filter (116.12s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/basic (32.21s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/update (26.99s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/tags (42.16s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/disappears (14.76s)
```

File: aws/resource_aws_guardduty_filter_test.go (hunk 1 above)
Comment (YakDriver) [reply aws/resource_aws_guardduty_filter_test.go:260]: Completed

**Review by bflad — APPROVED** [review]
You rock 🚀 

Output from acceptance testing in AWS Commercial:

```
    --- PASS: TestAccAWSGuardDuty_serial/Filter (92.19s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/basic (26.18s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/update (21.68s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/tags (32.22s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/disappears (12.11s)
```

Output from acceptance testing in AWS GovCloud (US):

```
    --- PASS: TestAccAWSGuardDuty_serial/Filter (122.38s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/disappears (17.71s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/basic (31.44s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/update (28.86s)
        --- PASS: TestAccAWSGuardDuty_serial/Filter/tags (44.37s)
```

**Pull request merged**